### ISSUE TITLE
OSX fixes: 6l -s, tell asciidoc which Python to use and where dot is at. [phab:D319]

### ISSUE THREAD

**Issue by kythebot** [google/kythe#539]
_Created by **zarko** at 2015-06-04 20:40:28_: 

On OS X, asciidoc fails to find the executable of the
Python interpreter, leading it to try and execute the
empty string. We can fix this by overriding the
autodetected interpreter with an attribute.

The environment Bazel runs builds in does not include
the path for graphviz on OS X. This CL adds logic to
configure this in setup_bazel.sh.

Finally, there's a Go issue on OS X that we're hitting.
(https://github.com/golang/go/issues/10254). Stop hitting it.

**Comment by kythe** [comment]
_**zarko** wrote at 2015-06-04 20:42:02_:

  - tabs are bad

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-06-04 20:43:46_:

I don't like editing the actual rule code.  It should be configured externally.

Why can't we assume `dot` is on `PATH`?

**Comment by kythe** [comment]
_**schroederc** required changes at 2015-06-04 20:43:46_

**Comment by kythe** [comment]
_**zarko** wrote at 2015-06-04 21:08:25_:

  - dot is really in the PATH

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2015-06-04 21:08:29_:

I had thought this was a separate issue than #512 but that hasn't landed yet. Looks like we can assume we can get `dot` from `command -v`.

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-06-04 21:10:35_:

I don't think we need this since `command -v` succeeded.

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2015-06-04 21:13:09_:

Removing this causes the build to fail.


```
INFO: From Executing genrule //kythe/docs:kythe-storage:
Traceback (most recent call last):
  File "/usr/local/Cellar/asciidoc/8.6.9/etc/asciidoc/filters/graphviz/graphviz2png.py", line 168, in <module>
    app = Application()
  File "/usr/local/Cellar/asciidoc/8.6.9/etc/asciidoc/filters/graphviz/graphviz2png.py", line 68, in __init__
    format_output = subprocess.Popen(["dot", "-T?"], stderr=subprocess.PIPE, stdout=subprocess.PIPE).communicate()[1]
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 709, in __init__
    errread, errwrite)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 1326, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
asciidoc: WARNING: kythe-storage.txt: line 106: filter non-zero exit code: "/usr/bin/python" "/usr/local/Cellar/asciidoc/8.6.9/etc/asciidoc/filters/graphviz/graphviz2png.py"  -o "/private/var/tmp/_bazel_zarko/4047a92cde5c126a2747538e6c63dc38/kythe/bazel-out/local_darwin-fastbuild/bin/kythe/docs/kythe-storage__1.png" -L dot -F png -: returned 1
asciidoc: WARNING: kythe-storage.txt: line 106: no output from filter: "/usr/bin/python" "/usr/local/Cellar/asciidoc/8.6.9/etc/asciidoc/filters/graphviz/graphviz2png.py"  -o "/private/var/tmp/_bazel_zarko/4047a92cde5c126a2747538e6c63dc38/kythe/bazel-out/local_darwin-fastbuild/bin/kythe/docs/kythe-storage__1.png" -L dot -F png -
ERROR: /Users/zarko/kythe/tools/build_rules/doc.bzl:13:3: Executing genrule //kythe/docs:kythe-storage failed: bash failed: error executing command /bin/bash -c ... (remaining 1 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
```

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-06-04 21:14:34_:

Can you just write `export PATH`?

**Comment by kythe** [comment]
_**zarko** wrote at 2015-06-04 21:26:53_:

  - Further simplify doc.bzl

**Comment by kythe** [comment]
_**schroederc** accepted this revision at 2015-06-04 21:28:05_

**Comment by kythe** [comment]
_**zarko** closed via commit 2406989ebdd62c7409d5f28d523a104cd30d69e6 at 2015-06-04 21:32:47_
